tests/conftest.py

```diff
@@ -144,3 +148,101 @@ def _generate_calculation_node(process_state=ProcessState.FINISHED, exit_status=
         return node
 
     return _generate_calculation_node
+
+
+@pytest.fixture
+def create_empty_config_instance() -> Config:
+    """Create a temporary configuration instance.
+
+    This creates a temporary directory with a clean `.aiida` folder and basic configuration file. The currently loaded
+    configuration and profile are stored in memory and at are automatically restored at the end of this context manager.
+
+    :return: a new empty config instance.
+    """
+    import tempfile
+
+    from aiida.common.utils import Capturing
+    from aiida.manage import configuration
+    from aiida.manage.configuration import settings, load_profile, reset_profile
+
+    with tempfile.TemporaryDirectory() as dirpath:
+
+        # Store the current configuration instance and config directory path
+        current_config = configuration.CONFIG
+        current_config_path = current_config.dirpath
+        current_profile_name = configuration.PROFILE.name
+
+        reset_profile()
+        configuration.CONFIG = None
+
+        # Create a temporary folder, set it as the current config directory path and reset the loaded configuration
+        settings.AIIDA_CONFIG_FOLDER = dirpath
+
+        # Create the instance base directory structure, the config file and a dummy profile
+        settings.create_instance_directories()
+
+        # The constructor of `Config` called by `load_config` will print warning messages about migrating it
+        with Capturing():
+            configuration.CONFIG = configuration.load_config(create=True)
+
+        try:
+            yield get_config()
+        finally:
```

Comment:
Its not generally necessary to wrap `yield` in a `try/finally` block with pytest fixtures: https://docs.pytest.org/en/5.4.3/fixture.html#fixture-finalization-executing-teardown-code